python/link_prediction.py

```diff
@@ -0,0 +1,771 @@
+
+        # Check typing
+        if type(metrics) != tuple:
+            return False, "metrics must be an iterable object. "
+
+        for metric in metrics:
+            _metric = metric.lower()
+            if (
+                _metric != "loss"
+                and _metric != "accuracy"
+                and _metric != "f1"
+                and _metric != "auc_score"
+                and _metric != "precision"
+                and _metric != "recall"
+                and _metric != "specificity"
+                and _metric != "num_wrong_examples"
+            ):
+                return False, "Metric name " + _metric + " is not supported!"
+
+    # Predictor type
+    if "predictor_type" in parameters.keys():
+        predictor_type = parameters["predictor_type"]
+
+        # Check typing
+        if type(predictor_type) != str:
+            return False, "predictor_type must be a string. "
+
+        if predictor_type != DOT_PREDICTOR and predictor_type != MLP_PREDICTOR:
+            return False, "Predictor " + predictor_type + " is not supported. "
+
+    # Attention heads
+    if "attn_num_heads" in parameters.keys():
+        attn_num_heads = parameters["attn_num_heads"]
+
+        if layer_type == GRAPH_ATTN:
+
+            # Check typing
+            if type(attn_num_heads) != tuple:
+                return False, "attn_num_heads must be an iterable object. "
+
+
+            if len(attn_num_heads) != len(hidden_features_size):
+                return (
+                    False,
+                    "Specified network with {} layers but given attention heads data for {} layers. ".format(
+                        len(hidden_features_size) - 1, len(attn_num_heads)
+                    ),
+                )
+
+            if attn_num_heads[-1] != 1:
+                return False, "Last GAT layer must contain only one attention head. "
+
+            for num_heads in attn_num_heads:
+                if num_heads <= 0:
+                    return (
+                        False,
+                        "GAT allows only positive, larger than 0 values for number of attention heads. ",
+                    )
+
+    # Training accuracy patience
+    if "tr_acc_patience" in parameters.keys():
+        tr_acc_patience = parameters["tr_acc_patience"]
+
+        # Check typing
+        if type(tr_acc_patience) != int:
+            return False, "tr_acc_patience must be an iterable object. "
+
+        if tr_acc_patience <= 0:
+            return False, "Training acc patience flag must be larger than 0."
+
+    # model_save_path
+    if "model_save_path" in parameters.keys():
+        model_save_path = parameters["model_save_path"]
+
+        # Check typing
+        if type(model_save_path) != str:
+            return False, "model_save_path must be a string. "
+
+        if model_save_path == "":
+            return False, "Path must be != " " "
+
+    return True, "OK"
+
+
+def _reset_train_predict_parameters() -> None:
```

Comment:
Create a Reset function so user can reset the parameters of training

Reply:
I was thinking about that, but I couldn't think of when this would be useful. E.g, if the user doesn't want to use these parameters, he/she can simply retrain the model and the parameters will then be reset. And I don't see any other thing the user could do without first retraining the model.

Reply:
Well there might be a case where someone would like to train on top of the current model's parameters, which is currently not possible because everything is reset at the beginning of training. Having an explicit function also makes it more understandable for user. We can discuss because both of us could be wrong on this one.

Reply:
I understand, I was going with the assumption that it shouldn't be allowed to train on top of the current model's parameters because of the existence of the validation set. So from the overfitting perspective, it makes sense to forbid retraining but from the underfitting perspective clearly not.